### PR TITLE
Clarify relative to terms defined in Java API docs and JLS 

### PR DESCRIPTION
**Pull request by elharo**
@garrettjonesgoogle @suztomo 

1. errors are not exceptions
2. throws clause, not declaration,
3. signature is redundant

### CODE REVIEW

File: library-best-practices/glossary.md [hunk 1]
```diff
@@ -37,9 +38,9 @@ Java Dependency Glossary
 <a name="linkage-conflict"></a>
 <a name="static-linkage-conflict"></a>
 <a name="dynamic-linkage-conflict"></a>
-- **Linkage conflict**: a linkage error when the signature, return type,
-  modifiers, or throws declaration of a non-private method, field, or class
-  in a dependency has changed (or removed) in an incompatible way between
+- **Linkage conflict**: a linkage error when the name, return type,
+  modifiers, arguments, or throws clause of a non-private method, field, or class
```
Comment (suztomo) [inline library-best-practices/glossary.md]: I found a change to `throws` clause does not break compatibility. Shall we remove this part?

> Changes to the throws clause of methods or constructors do not break compatibility with pre-existing binaries; these clauses are checked only at compile time.
https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html#jls-13.4.21

File: library-best-practices/glossary.md [hunk 2]
```diff
@@ -19,50 +19,26 @@ Java Dependency Glossary
 ### Types of conflicts and compatibility
 
 <a name="linkage-error"></a>
-<a name="static-linkage-error"></a>
-<a name="dynamic-linkage-error"></a>
-- **Linkage error**: an error when a Java class references
-  another class and the reference cannot be satisfied with the available classes in the class path.
+- **Linkage error**: A "class has some dependency on another class; however, the
```
Comment (suztomo) [inline library-best-practices/glossary.md]: I felt the double quotation typo.
image

Also would you define this as a noun, not a quoted sentence? For example "an error when ..."

Comment (suztomo) [reply library-best-practices/glossary.md]: How about 

> An unsatisfactory dependency where "class has some dependency on another ....

?

Comment (elharo) [reply library-best-practices/glossary.md]: fixed

**Pull request merged**